src/a-runtime-console/kubernetes/ui/pipeline/full-history-page/full-history-page.pipeline.spec.ts

```diff
@@ -1,64 +1,55 @@
-import { KubernetesComponentsModule } from '../../../components/components.module';
-import { KubernetesStoreModule } from '../../../kubernetes.store.module';
-import { BuildConfigDialogsModule } from '../../buildconfig/delete-dialog/buildconfig.dialogs.module';
-import { BuildStageViewComponent } from '../build-stage-view/build-stage-view.component';
-import { PipelinesFullHistoryToolbarComponent } from '../full-history-toolbar/full-history-toolbar.pipeline.component';
-import { PipelinesFullHistoryComponent } from '../full-history/full-history.pipeline.component';
-import { TestAppModule } from './../../../../app.test.module';
-import { StageTimePipe } from './../build-stage-view/stage-time.pipe';
 import { PipelinesFullHistoryPage } from './full-history-page.pipeline.component';
 
-import { StackDetailsModule } from 'fabric8-stack-analysis-ui';
-import { InputActionDialog } from '../input-action-dialog/input-action-dialog.component';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { APIsStore } from '../../../store/apis.store';
+import { BuildStore } from '../../../store/build.store';
+import { BuildConfigStore } from '../../../store/buildconfig.store';
+
+import { BuildService } from '../../../service/build.service';
+import { BuildConfigService } from '../../../service/buildconfig.service';
+import { DevNamespaceScope, TestDevNamespaceScope } from '../../../service/devnamespace.scope';
 
 describe('PipelinesFullHistoryPage', () => {
   let component: PipelinesFullHistoryPage;
   let fixture: ComponentFixture<PipelinesFullHistoryPage>;
 
   beforeEach(async(() => {
+    let mockBuildConfigService: any = jasmine.createSpy('BuildConfigService');
+    let mockDevNameSpaceScrope: any = jasmine.createSpy('DevNamespaceScope');
+    let mockBuildService: any = jasmine.createSpy('BuildService');
+    let mockAPIsStore: any = jasmine.createSpy('APIsStore');
     TestBed.configureTestingModule({
-      imports: [
-        Fabric8CommonModule,
-        RouterTestingModule.withRoutes([]),
-        RestangularModule.forRoot(),
-        FormsModule,
-        MomentModule,
-        ModalModule,
-        KubernetesStoreModule,
-        KubernetesComponentsModule,
-        BuildConfigDialogsModule,
-        TestAppModule,
-        StackDetailsModule
-      ],
       declarations: [
-        BuildStageViewComponent,
-        InputActionDialog,
-        PipelinesFullHistoryPage,
-        PipelinesFullHistoryComponent,
-        PipelinesFullHistoryToolbarComponent,
-        StageTimePipe
+        PipelinesFullHistoryPage
       ],
       providers: [
-        MockBackend,
-        { provide: RequestOptions, useClass: BaseRequestOptions },
         {
-          provide: Http, useFactory: (backend, options) => {
-            return new Http(backend, options);
-          }, deps: [MockBackend, RequestOptions]
-        }
-      ]
-    })
-      .compileComponents();
+          provide: BuildConfigStore, useFactory: (
```

Comment:
@jiekang The PhantomJS crash issue is solved now. Only 3 tests are failing currently because test context is not properly set for them. I tried different ways to mock the services used in the constructor but somehow it seems too complex to me. Maybe i am missing something here and there is a simple way to mock the methods.

Reply:
@rohitkrai03 I'm going to open a PR against your branch (aka "PR your PR") with corrections for these test mocks. Then Jie and I can come back around and review the actual changes without the noise of spurious test failures.